frontend/src/features/workflow/DatasetPreview.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react'
+import { Dataset, ComponentList, Structure, Body } from '../../qri/dataset'
+
+interface DatasetPreviewProps {
+  data: Dataset
+}
+
+export const DatasetPreview: React.FC<DatasetPreviewProps> = ({ data }) => {
+  const [componentTab, setComponentTab] = useState('commit')
+
+  const presentComponents: string[] = Object.keys(data).filter((c: string) => {
+    return ComponentList.includes(c) && c !== 'body'
+  })
+
+  const tabs: Tab[] = []
+  ComponentList.forEach((component: string) => { 
+    if (component === 'body') {
+      return
+    }
+    const tab: Tab = { name: component }
+    if (presentComponents.includes(component)) {
+      tab.clickable = true
+    }
+    tabs.push(tab)
+  })
+
+  return <div className='w-full bg-white'>
+    <TabbedDisplay
+      activeTab={componentTab}
+      tabs={tabs}
+      onTabClick={setComponentTab}
+      content={data[componentTab]}
+    />
+    <BodyDisplay 
+      structure={data.structure}
+      body={data.body}
+    />
+  </div>
+}
+
+interface Tab {
+  name: string
+  clickable?: boolean
+}
+
+interface TabbedDisplayProps {
+  activeTab: string
+  tabs: Tab[]
+  onTabClick: (tab: string) => void
+  content: string | Record<string, any>
+}
+
+const TabbedDisplay: React.FC<TabbedDisplayProps> = (props) => {
+  const {
+    activeTab,
+    tabs,
+    onTabClick,
+    content
+  } = props
+
+  return <div className='w-full'>
```

Comment:
I'm fully bought in on the cool-kid way of writing this. Sorry for flip-flopping.
```suggestion
const TabbedDisplay: React.FC<TabbedDisplayProps> = ({
    activeTab,
    tabs,
    onTabClick,
    content
}) => (
  <div className='w-full'>
  {/* ... */}
```